shap/explainers/tree.py

```diff
@@ -245,9 +245,10 @@ def shap_values(self, X, y=None, tree_limit=None, approximate=False, check_addit
                 assert not approximate, "approximate=True is not supported for LightGBM models!"
                 phi = self.model.original_model.predict(X, num_iteration=tree_limit, pred_contrib=True)
                 # Note: the data must be joined on the last axis
-                if self.model.original_model.params['objective'] == 'binary':
-                    warnings.warn('LightGBM binary classifier with TreeExplainer shap values output has changed to a list of ndarray')
-                    phi = np.concatenate((0-phi, phi), axis=-1)
+                if self.model.original_model.params.get('objective') is not None:
```

Comment:
can you add a test for this, I can't repro this issue.  I also sent a PR here, but the test test_lightgbm_mse_regressor doesn't actually fail when I undo the fix, so I am a bit confused as to why it is failing on some environments:
https://github.com/slundberg/shap/pull/1063

Reply:
also I'm confused by this description:

```
But this does not create an 'objective' entry in the params dictionary unless explicitly specified
```

If that was the case, then the test "test_lightgbm" under test_tree.py would always fail...

Reply:
Let me look into that when i'm back from work. As for reproducing the error, maybe the issue is I wasn't using LGBMRegressor, but instead called directly the .train() function.
Something like:

`import lightgbm as lgb`

`lgb_train = lgb.Dataset(x_train,y_train)`

`lgb_test = lgb.Dataset(x_test,y_test, reference=lgb_train)`

`params = {
    'num_threads': 4,
    'n_estimators': 8000,
    'early_stopping_round':50, 
    'metric': ['rmse'],
    'random_state': C_SEED,
    'verbose': 1,
}`

`evals_result = {}  `

`gbm = lgb.train(params,
                lgb_train,
                valid_sets=[lgb_train, lgb_test],
                evals_result=evals_result,
                verbose_eval=30
               )
`

I'm new to this library, are both interchangeable if objective is regression?

Reply:
@ggmblr thank you, I added your suggested code and was able to 1.) reproduce the issue 2.) validate the issue is fixed with the code changes

Reply:
https://github.com/slundberg/shap/pull/1063